pycheribuild/projects/syzkaller.py

```diff
@@ -62,92 +68,128 @@ def setup_config_options(cls, **kwargs):
             "run-sysgen", show_help=True,
             help="Rerun syz-extract and syz-sysgen to rebuild generated Go "
                  "syscall descriptions.")
+        cls.use_go = cls.add_config_option("use-go", kind=GoType, default=GoType.SYSTEM,
+                                           enum_choice_strings=[t.value for t in GoType],
+                                           help="The Go type to run with.")
+
+    def get_path(self):
+        return (str(self.get_sdk_bindir()) + ":" +
+                str(self.config.dollar_path_with_other_tools))
+
+    def get_sdk_dir(self):
+        return self.config.cheri_sdk_dir
+
+    def get_sdk_bindir(self):
+        return self.config.cheri_sdk_bindir
 
     def __init__(self, config):
-        self._install_prefix = config.cheri_sdk_dir
-        self._install_dir = config.cheri_sdk_dir
-        self.destdir = ""
         super().__init__(config)
+        self._install_prefix = self.get_sdk_dir()
+        self._install_dir = self.get_sdk_dir()
+        self.destdir = ""
 
         # self.gopath = source_base / gohome
-        self.goroot = config.cheri_sdk_dir / "go"
+        self.goroot = self.get_sdk_dir() / "go"
 
         # repo_url = urlparse(self.repository.url)
         # repo_path = repo_url.path.split(".")[0]
         # parts = ["src", repo_url.netloc] + repo_path.split("/")
         self.gopath = self.build_dir
         self.gosrc = self.source_dir
 
-        self._new_path = (str(self.config.cheri_sdk_dir / "bin") + ":" +
-                          str(self.config.dollar_path_with_other_tools))
+        self.targetarch = self.crosscompile_target.cpu_architecture.value
+        if self.targetarch == "aarch64":
+            self.targetarch = "arm64"
 
         cheribsd_target = self.get_crosscompile_target(config).get_rootfs_target()
         self.cheribsd_dir = BuildCHERIBSD.get_source_dir(self, cross_target=cheribsd_target)
 
+        self._setup_make_args()
+
     def syzkaller_install_path(self):
-        return self.config.cheri_sdk_bindir
+        return self.get_sdk_dir() / "syzkaller"
 
     def syzkaller_binary(self):
-        return self.config.cheri_sdk_bindir / "syz-manager"
+        return self.get_sdk_bindir() / "syz-manager"
 
     def needs_configure(self) -> bool:
         return False
 
+    def _setup_make_args(self):
```

Comment:
```suggestion
    def setup(self):
        super().setup()
```

The setup() function will be called automatically after __init__(), so no need to call it from there.
Any setup logic that doesn't set instance variables should ideally be moved to setup(). __init__() is always called, but setup() is only called when the target is actually executed, so moving as much as possible to setup() reduces startup time (and things like tab-completion).